@Test public void setParameterWithNullParameterThrowsException() throws SQLException {
  doThrow(new SQLException("invalid column")).when(ps).setNull(1,JdbcType.INTEGER.TYPE_CODE);
  try {
    TYPE_HANDLER.setParameter(ps,1,null,JdbcType.INTEGER);
    Assert.fail("Should have thrown a TypeException");
  }
 catch (  Exception e) {
    Assert.assertTrue("Expected TypedException",e instanceof TypeException);
    Assert.assertTrue("Parameter index is in exception",e.getMessage().contains("parameter #1"));
  }
}
